### ISSUE TITLE
[K8s] list_pod_for_all_namespaces gives ApiException: (403) if the user doesn't have necessary permissions

### ISSUE THREAD

**Issue by hemildesai** [skypilot-org/skypilot#4276]
The error looks like
```
pods is forbidden: User \"...\" cannot list resource \"pods\" in API group \"\" at the cluster scope
```

In this case, the kubernetes client should fall back to `list_namespaced_pod` and use default context namespace.

Ref: https://github.com/skypilot-org/skypilot/blob/master/sky/provision/kubernetes/utils.py#L399 and https://github.com/skypilot-org/skypilot/blob/master/sky/provision/kubernetes/utils.py#L2019

_Version & Commit info:_
* `sky -v`: 0.7.0
* `sky -c`: 3f625886bf1b13ee463a9f8e0f6741f620f7f66f

**Comment by hemildesai** [comment]
I'm happy to submit a PR to fix this. For me locally, the following change works
```python
try:
    pods = kubernetes.core_api(context).list_pod_for_all_namespaces(
        _request_timeout=kubernetes.API_TIMEOUT).items
except kubernetes.api_exception():
    namespace = get_kube_config_context_namespace(context)
    pods = kubernetes.core_api(context).list_namespaced_pod(namespace).items
```

**Comment by hemildesai** [comment]
> Hey @hemildesai! We'll have this fixed in https://github.com/skypilot-org/skypilot/pull/4208 :)

Awesome, thanks 🎉

**Comment by romilbhardwaj** [comment]
Should be fixed in #4208.
